### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by savente93**
## Issue addressed
Fixes NA

## Explanation
I noticed that our current test suite had some warnings so I decided to fix them in a separate PR. I actually think that warnings are important to pay attention to since they do often indicate problems. I've fixed them now for the current test suite and if it gets merged I would like to argue for trying to keep the test suite warning free going forward. Of course, sometimes there are issues we don't have control over, and in that case we can add them to the filters in the `pyproject.toml` but then at least we're making concussions decisions.

## Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `v1`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation if needed
- [x] Updated changelog.rst if needed
- [x] For predefined catalogs: update the catalog version in the file itself, the references in data/predefined_catalogs.yml, and the changelog in data/changelog.rst

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

File: hydromt/gis/raster.py [hunk 1]
```diff
@@ -2290,9 +2290,9 @@ def interpolate_na(
         kwargs.update(dict(method=method, extrapolate=extrapolate))
         if dim0:
             interp_data = np.empty(self._obj.shape, dtype=self._obj.dtype)
-            for i, (_, sub_xds) in enumerate(self._obj.groupby(dim0)):
+            for i, (_, sub_xds) in enumerate(self._obj.groupby(dim0, squeeze=False)):
```
Comment (Jaapel) [inline hydromt/gis/raster.py:2293]: What is this doing?

Comment (savente93) [reply hydromt/gis/raster.py:2293]: Squeeze removes empty dimensions. Currently the default is that it sill squeeze, but according to the error, this is being removed, so instead I set squeeze to False (which is the only way to silence the warning) and call squeeze on the result to maintain the same behavior.

**Comment by Jaapel** [comment]
Does the migration guide need to be updated?

**Comment by savente93** [comment]
Not for this one I think since there are no changes necessary on the part of the user. I have updated the changelog, but I don't think the migration guide needs anything

**Pull request merged**